[WIP] Refactor Migration Guides (#386)
fluxcd/website: content/en/docs/migration/faq-migration.md

On top of that, testing the individual components and understanding the codebase becomes a lot easier.

### Are there any breaking changes?

> - In Flux v1 Kustomize support was implemented through `.flux.yaml` files in the Git repository. As indicated in the comparison table above, while this approach worked, we found it to be error-prone and hard to debug. The new [Kustomization CR](https://github.com/fluxcd/kustomize-controller/blob/master/docs/spec/v1beta1/kustomization.md) should make troubleshooting much easier. Unfortunately we needed to drop the support for custom commands as running arbitrary shell scripts in-cluster poses serious security concerns.

Reviewer: I like the language in this change better. Maybe still explicitly mention `.flux.yaml` in there so people know which feature this refers to?

> After experimentation and considerable thought, we (the maintainers) have found a path to Flux v2 that we think better serves our vision of GitOps: the GitOps Toolkit. In consequence, we do not now plan to integrate GitOps Engine into Flux.

Author: ```suggestion
After experimentation and considerable thought, we (the maintainers) have found a path to Flux v2 that better serves our vision of GitOps: the GitOps Toolkit. As a result, we do not plan to integrate GitOps Engine into Flux.
```

nit : language

Not yet, but the feature is under active development. See the [image update feature parity section on the roadmap](/docs/roadmap/#flux-image-update-feature-parity) for updates on this topic.

### How do I automatically apply my `HelmRelease` resources to the cluster?

If you are currently a Flux v1 user, you can commit the `HelmRelease` resources to Git, and Flux will automatically apply them to the cluster like any other resource. It does however not support automated image updates for Helm Controller resources.

If you are not a Flux v1 user or want to fully migrate to Flux v2, the [Kustomize Controller](/docs/components/kustomize/controller/) will serve your needs.

### I am still running Helm v2, what is the right upgrade path for me?

Migrate your Helm v2 releases to v3 using [the Helm Operator's migration feature](/legacy/helm-operator/helmrelease-guide/release-configuration/#migrating-from-helm-v2-to-v3), or make use of the [`helm-2to3`](https://github.com/helm/helm-2to3) plugin directly, before continuing following the [migration steps](#steps).

### Is the Helm Controller ready for production?

Probably, but with some side notes:

> 1. It is still under active development, and while our focus has been to stabilize the API as much as we can during the first development phase, we do not guarantee there will not be any breaking changes before we reach General Availability. We are however committed to provide [conversion webhooks](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#webhook-conversion) for upcoming API versions.

Reviewer: Aren't we guaranteeing now that APIs will stay stable? https://fluxcd.io/docs/components/helm/api/
